### PR TITLE
rewrite addPodStorage

### LINKED ISSUE #334: Dashboard `Your storage` do not display anything
This is as from v3.5.16 of solid-panes.
The problem do not arise when building mashlib with `watch`. 

related code is in src/outliner/manager.js

### PR DESCRIPTION
**Pull request by bourgeoa**
resolves issue #334 
Tested locally on a multi user localhost as databrowser and web app.

### CODE REVIEW

File: src/outline/manager.js [hunk 1]
```diff
@@ -442,16 +442,32 @@ export default function (context) {
...
+      // remove namedNodes duplicates
+      function uniques (nodes) {
+        const uniqueNodes = []
+        nodes.forEach(node => {
+          if (!uniqueNodes.find(uniqueNode => uniqueNode.equals(node))) uniqueNodes.push(node)
+        })
+        return uniqueNodes
+      }
+      pods = uniques(pods)
+      console.log('return pods')
```
Comment (SharonStrats) [inline src/outline/manager.js]: suggestion: if it's not too much trouble I would use debug instead of console.log.

File: src/outline/manager.js [hunk 2]
```diff
@@ -416,23 +416,23 @@ export default function (context) {
...
-        if (!pods.some(pod => pod.uri.includes(podStorage.origin))) {
-          // check in order, location.origin/<podName> then location.origin containing space:Storage
-          if (podStorage.pathname === '/' || !(await addPodRoot(kb.sym(`${podStorage.origin}/${podStorage.pathname.split('/')[1]}/`)))) {
-            await addPodRoot(kb.sym(`${podStorage.origin}/`))
-          }
+        // check for predicate pim:Storage in containers up the path tree
+        let pathStorage = podStorage.pathname
+        while (pathStorage.length) {
+          pathStorage = pathStorage.substring(0, pathStorage.lastIndexOf('/'))
+          if (await addPodStorage(kb.sym(`${podStorage.origin}${pathStorage}/`))) return
```
Comment (SharonStrats) [inline src/outline/manager.js:433]: question: I'm curious to know the difference between the storage paths where there is an index.html.  Also, why the change from root to storage.  Will be moving away from root and have a separate storage "directory"?

**Review by SharonStrats — APPROVED** [review]
Very cool.  I haven't yet gotten solidos to run.  Hoping to tonight, if so I will also test this locally.

File: src/outline/manager.js (hunk 2 above)
Comment (bourgeoa) [reply src/outline/manager.js:433]: The previous code was not loading the folder representation when the authentification was done from a window.location.href that was not the podRoot. This situation occurs always for pod with index.html at root.

Comment (SharonStrats) [reply src/outline/manager.js:433]: I have tested it using your profile link https://bourgeoa.solidcommunity.net:8443/profile/card#me, then logging in as me (my test server account) clicking storage.  It works for me. I can click between your storage and mine as well and it works very well!  Nice job!

File: src/outline/manager.js (hunk 1 above)
Comment (bourgeoa) [reply src/outline/manager.js]: removed
